synapse/rest/key/v2/local_key_resource.py

```diff
@@ -100,6 +102,12 @@ def response_json_object(self) -> JsonDict:
     def on_GET(
         self, request: Request, key_id: Optional[str] = None
     ) -> Tuple[int, JsonDict]:
+        # Matrix 1.6 drops support for passing the key_id, this is incompatible
+        # with earlier versions and is allowed in order to support both.
+        # A warning is issued to help determine when it is safe to drop this.
+        if key_id:
+            logger.warning("Request received for local key with key ID: %s", key_id)
```

Comment:
For people scrolling through logs, it would be nice to include more context in the warning why we even consider this a warning so they don't have to follow red herrings (applies to the log below).

```suggestion
        # Matrix 1.6 drops support for passing the key_id (per MSC3938). This is incompatible
        # with earlier versions and is allowed in order to support both.
        # A warning is issued to help determine when it is safe to drop this.
        if key_id:
            logger.warning("Request received for local key with key ID which is deprecated (logging to determine level of usage so we can remove in the future): %s", key_id)
```

Reply:
Thanks! I tweaked it slightly to make it shorter, but I think I capture the same sentiment.

Reply:
Updated logs look good :+1: 

The MSC was certainly the easiest way to onboard onto this context :shrug: 

I don't even know how to see what changed in v1.6 -> https://spec.matrix.org/v1.6/

I mean I see the [blog post](https://matrix.org/blog/2023/02/14/matrix-v-1-6-release) but that's only because you linked it in https://github.com/matrix-org/synapse/issues/15089 which this PR is addressing. If I'm coming at this from the comment in the code, I don't know how to easily find this info.

Reply:
Every spec version has a changelog which is essentially the blog post: https://spec.matrix.org/v1.6/changelog/

Reply:
Looks sufficient! It still takes some searching through but it's there :+1: . Thanks for sharing the link!

Reply:
No problem! I really try *not* to reference back to MSCs since theoretically all the info should be in the spec, but ... yeah that doesn't always happen. 😢